haskell/experimental/transitions.bzl

```diff
@@ -0,0 +1,16 @@
+load("//haskell:private/pkg_id.bzl", "pkg_id")
+
+def _package_name_out_transition_impl(settings, attr):
+    package_name = getattr(attr, "package_name", None)
+    version = getattr(attr, "version", None)
+    # we don't have direct access to ctx.label here, so we've recreated it in
+    # "attr.label". See //haskell:defs.bzl
```

Comment:
Ah, yes, that's tricky. Sadly, I don't have a better idea how to deal with this either. Nice workaround!